src/locale/localed.c

```diff
@@ -302,7 +304,7 @@ static int method_set_locale(sd_bus_message *m, void *userdata, sd_bus_error *er
                         k = strlen(name);
                         if (startswith(*i, name) &&
                             (*i)[k] == '=' &&
-                            locale_is_valid((*i) + k + 1)) {
+                            locale_is_installed((*i) + k + 1) <= 0) {
```

Comment:
The logic is inverted here. `<0` should cause a failure, and `>0` should be considered installed.

I think this code would be clearer if this whole internal loop was split out to a helper... We'd avoid the need to set a flag variable then.